src/protocol/attribution/credit_capping.rs

```diff
@@ -482,4 +482,76 @@ mod tests {
             assert_eq!(v.as_u128(), *expected);
         }
     }
+
+    #[tokio::test]
+    #[should_panic]
+    pub async fn wrapping_add_attack() {
+        const CAP: u32 = 2;
+        const NUM_MULTI_BITS: u32 = 3;
+
+        // Demonstration of the wrapping-add attack.
+        // Prefix-sum with the following input yields (5 + 26) = 0 in Fp31.
+        const EXPECTED: &[u128; 4] = &[2, 0, 0, 0];
+        let input: Vec<GenericReportTestInput<Fp31, MatchKey, BreakdownKey>> = accumulation_test_input!(
+            [
+                { is_trigger_report: 0, helper_bit: 0, breakdown_key: 0, credit: 5 },
+                { is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 5 },
+                { is_trigger_report: 0, helper_bit: 1, breakdown_key: 1, credit: 26 },
+                { is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 26 },
+            ];
+            (Fp31, MatchKey, BreakdownKey)
+        );
```

Comment:
I'd prefer to see the original IPA input. This type of input is tightly coupled to the implementation details of the attribution stage.